runtime-modules/council/src/lib.rs

```diff
@@ -527,6 +607,68 @@ impl<T: Trait> Module<T> {
         Self::deposit_event(RawEvent::AnnouncingPeriodStarted());
     }
 
+    /////////////////// Budget-related /////////////////////////////////////
+
+    /// pay rewards to elected council members
+    fn pay_elected_member_rewards(now: T::BlockNumber) {
+        let reward_per_block = T::ElectedMemberRewardPerBlock::get();
+
+        // intermediate balance
+        let mut new_balance = Budget::<T>::get();
```

Comment:
I think you can sidestep this mutable with a map+reduce type pattern here, but I think we can skip it for now, this feels relatively manageable.

Reply:
I usually try to get rid of side-effects, but the reason I loop here via `for` instead of reducing the vector was the easy use of circuit-shorting (`break` out of `for`) when budget is completely depleted.

Reply:
In the end, `for` loop was removed in favor of `fold()`(reduce) due to rework mention here https://github.com/Joystream/joystream/pull/1760#discussion_r527630479 .